src/redux/books/books.js

```diff
@@ -1,28 +1,68 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { addAPI, fetchAPI, removeAPI } from '../../API/apis';
+
+const FETCH_BOOKS = 'bookstore/books/FETCH_BOOKS';
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
 
-const initState = [];
-
-export const addBook = (payload) => ({
-  type: ADD_BOOK,
-  payload,
-});
+const fetchBooks = createAsyncThunk(
+  FETCH_BOOKS,
+  async () => {
+    const response = await fetchAPI();
+    return response;
+  },
+);
 
-export const removeBook = (payload) => ({
-  type: REMOVE_BOOK,
-  payload,
-});
+const addBook = createAsyncThunk(
+  ADD_BOOK,
+  async (Obj) => {
+    await addAPI(Obj);
+    return Obj;
+  },
+);
 
-const booksReducer = (state = initState, action) => {
-  switch (action.type) {
-    case ADD_BOOK:
-      return [...state, action.payload];
+const removeBook = createAsyncThunk(
+  REMOVE_BOOK,
+  async (id) => {
+    await removeAPI(id);
+    return id;
+  },
+);
 
-    case REMOVE_BOOK:
-      return state.filter((book) => book.id !== action.payload);
-    default:
-      return state;
-  }
+const initialState = {
+  status: null,
+  entities: [],
 };
 
-export default booksReducer;
+const handleBookSlice = createSlice({
+  name: 'handleBook',
+  initialState,
+  extraReducers: (builder) => {
+    builder.addCase(fetchBooks.fulfilled, (state, action) => {
+      const newBookList = [];
+      Object.entries(action.payload).forEach((item) => {
+        newBookList.push({
+          id: item[0],
+          title: item[1][0].title,
+          author: item[1][0].author,
+        });
+      });
+      // eslint-disable-next-line no-param-reassign
+      state.entities = newBookList;
+    });
+    builder.addCase(addBook.fulfilled, (state, action) => {
+      state.entities.push({
+        id: action.payload.item_id,
+        title: action.payload.title,
+        author: action.payload.author,
+      });
+    });
+    builder.addCase(removeBook.fulfilled, (state, action) => {
+      // eslint-disable-next-line no-param-reassign
```

Comment:
- [ ] As I mentioned above, kindly make sure you fix your linters errors instead of dasabling them for best practices.

Reply:
- [ ] @blase147 great work fixing the changes requested, however, it's best practice to solve linter errors before disabling them, I would advise you to try and fix the linter error first.